tests/functional/spec/components/radio.spec.js

```diff
@@ -0,0 +1,236 @@
+
+     // Choose none option
+     .click(RadioMandatoryPage.none())
+     .isSelected(RadioMandatoryPage.none()).should.eventually.be.true
+
+     // Submit answers given on mandatory page
+     .click(RadioMandatoryPage.submit())
+
+     // None option should not be selected by default
+     .isSelected(RadioNonMandatoryPage.coffee()).should.eventually.be.false
+
+     // Choose None option
+     .click(RadioNonMandatoryPage.coffee())
+     .isSelected(RadioNonMandatoryPage.coffee()).should.eventually.be.true
+
+     // Submit answers given on non-mandatory page
+     .click(RadioNonMandatoryPage.submit())
+
+     // Then the answers are shown on the summary page
+     .getUrl().should.eventually.contain(SummaryPage.pageName)
+     .getText(SummaryPage.answer_mandatory()).should.eventually.contain('None')
+     .getText(SummaryPage.answer_optional()).should.eventually.contain('Coffee');
+    });
+  });
+
+it('Given when I update my answers the content on the summary page is updated', function() {
+  return helpers.openQuestionnaire(radio_schema).then(() => {
+    return browser
+      // Nothing should be selected by default
+      .isSelected(RadioMandatoryPage.none()).should.eventually.be.false
+      .isSelected(RadioMandatoryPage.bacon()).should.eventually.be.false
+      .isSelected(RadioMandatoryPage.eggs()).should.eventually.be.false
+      .isSelected(RadioMandatoryPage.sausage()).should.eventually.be.false
+      .isSelected(RadioMandatoryPage.other()).should.eventually.be.false
+
+      // Choose Eggs first
+      .click(RadioMandatoryPage.eggs())
+      .isSelected(RadioMandatoryPage.eggs()).should.eventually.be.true
+
+      // Submit answers given on mandatory page
+      .click(RadioMandatoryPage.submit())
+
+      // Skipping non-mandatory page
+      .click(RadioNonMandatoryPage.submit())
+
+       // Then my answer is shown
+      .getText(SummaryPage.answer_mandatory()).should.eventually.contain('Eggs')
+
+       // When I return to the previous screen
+      .click(SummaryPage.previous())
+
+      // Click non-mandatory option
+      .click(RadioNonMandatoryPage.tea())
+      .isSelected(RadioNonMandatoryPage.tea()).should.eventually.be.true
+
+      // Submit answers given on non-mandatory page
+      .click(RadioNonMandatoryPage.submit())
+
+      // Then the answers are shown on the summary page and updated to include non-mandatory answer
+      .getUrl().should.eventually.contain(SummaryPage.pageName)
+      .getText(SummaryPage.answer_mandatory()).should.eventually.contain('Eggs')
+      .getText(SummaryPage.answer_optional()).should.eventually.contain('Tea');
+    });
+  });
+
+it('Given the "other" option is available, when I clicks the "other" option then the other input should be visible', function() {
+     return helpers.openQuestionnaire(radio_schema).then(() => {
+      return browser
+        // When I click on the other option
+        .click(RadioMandatoryPage.other())
```

Comment:
Should we test (error presence?) for clicking "other" but providing no value in the other input field on the mandatory page?